lib/y2lan_restart_common.pm

```diff
@@ -86,21 +91,103 @@ sub check_network_status {
     if ($expected_status eq 'restart') {
         assert_script_run '[ -s journal.log ]';                       # journal.log size is greater than zero (network restarted)
     }
-    elsif (is_sle('<15') || is_leap('<15.0')) {
-        assert_script_run '[ ! -s journal.log ]';
-    }
+
     assert_script_run '> journal.log';                                # clear journal.log
     type_string "\n\n";                                               # make space for better readability of the console
 }
 
 sub verify_network_configuration {
-    my ($fn, $dev_name, $expected_status, $workaround) = @_;
+    my ($fn, $dev_name, $expected_status, $workaround, $no_network_check) = @_;
     open_network_settings;
 
     $fn->($dev_name) if $fn;                                          # verify specific action
 
     close_network_settings;
-    check_network_status($expected_status, $workaround);
+    check_network_status($expected_status, $workaround) unless defined $no_network_check;
+}
+
+sub validate_etc_hosts_entry {
+    my (%args) = @_;
+
+    script_run("egrep \"@{[$args{ip}]}\\s@{[$args{fqdn}]}\\s@{[$args{host}]}\" /etc/hosts", 30)
+      && record_soft_failure "bsc#1115644 Expected entry:\n \"@{[$args{ip}]}    @{[$args{fqdn}]} @{[$args{host}]}\" was not found in /etc/hosts";
+    script_run "cat /etc/hosts";
+}
+
+sub set_network {
+    my (%args) = @_;
+
+    open_network_settings;
+    send_key 'alt-i';    # edit NIC
+    assert_screen 'yast2_lan_network_card_setup';
+    if ($args{static}) {
+        send_key 'alt-t';    # set to static ip
+        assert_screen 'yast2_lan_static_ip_selected';
+        send_key 'tab';
+        if ($args{ip}) {     # To spare time, no update what to is already filled from previous run
+            send_key_until_needlematch('ip_textfield_empty', 'backspace');    # delete existing IP if any
+            type_string $args{ip};
+        }
+        send_key 'tab';
+        if ($args{mask}) {                                                    # To spare time, no update what to is already filled from previous run
+            send_key_until_needlematch('mask_textfield_empty', 'backspace');    # delete existing netmask if any
+            type_string $args{mask};
+        }
+        send_key 'tab';
+        send_key_until_needlematch('hostname_textfield_empty', 'backspace');
+        type_string $args{fqdn};
+        assert_screen 'yast2_lan_static_ip_set';
+    }
+    else {
+        send_key 'alt-y';                                                       # set back to DHCP
+        assert_screen 'yast2_lan_dhcp_set';
+    }
+    # Exit
+    send_key 'alt-n';
+    assert_screen "yast2_lan";
+    close_network_settings;
+}
+
+
+sub check_etc_hosts_update {
+
+=head2
```

Comment:
usually this section goes **before function name

Reply:
Right, that change is coming from the commits PR is based on, will fix that. Thanks!